### ISSUE TITLE
deprecation(log): deprecate `LogRecord` class and use plain object instead

### ISSUE THREAD

**Issue by timreichen** [denoland/std#4390]
**Is your feature request related to a problem? Please describe.**

`LogRecord` is a class with only properties and getters. It seems the same can be achieved with plain js object.

**Describe the solution you'd like**

Deprecate the `LogRecord` class and export `LogRecord` an interface instead. Replace all `new LogRecord()` occurrences with a plain object.
```ts
export interface LogRecord {
  msg: string;
  args: unknown[];
  datetime: Date;
  level: number;
  levelName: string;
  loggerName: string;
}
```

**Describe alternatives you've considered**

Leave as is.

**Comment by iuioiua** [comment]
I'm for this. Having a plainer/simpler symbol that achieves the same functionality is a good idea.

**Comment by timreichen** [comment]
I just wonder how to introduce this without a breaking change. Should we just deprecate the class and introduce the interface after the class removal?

**Comment by iuioiua** [comment]
I'm not sure there is a way to make a non-breaking change either. However, the change seems easy-to-make, from the user's perspective. Any thoughts, @kt3k?

Before:
```ts
...
fileHandler.handle(
  new LogRecord({
    msg: "AAA",
    args: [],
    level: LogLevels.ERROR,
    loggerName: "default",
  }),
);
...
```

After:
```ts
...
fileHandler.handle({
  msg: "AAA",
  args: [],
  level: LogLevels.ERROR,
  loggerName: "default",
});
```

**Comment by iuioiua** [comment]
@kt3k, would you be in favour of doing this in v1?

**Comment by kt3k** [comment]
I don't understand the motivation. It doesn't seem solving any problem.

> the same can be achieved with plain js object

I think this assumption is not strictly correct. The getter-only fields are currently immutable, but if we change it to plain object, they become mutable (`readonly` works for TypeScript, but it doesn't work for JavaScript).

**Comment by timreichen** [comment]
> I don't understand the motivation. It doesn't seem solving any problem.
> 
> > the same can be achieved with plain js object
> 
> I think this assumption is not strictly correct. The getter-only fields are currently immutable, but if we change it to plain object, they become mutable (`readonly` works for TypeScript, but it doesn't work for JavaScript).

That is true, but these properties have getters because the values are objects. So in the spirit of that whole class with getters and readonly properties, the object could just as well be frozen.

**Comment by kt3k** [comment]
We removed std/log in https://github.com/denoland/std/pull/6819

See the discussions in https://github.com/denoland/std/issues/6124
